### PR TITLE
Use gitpod for developers

### LINKED ISSUE #1835: Add a gitpod configuration for all
[Gitpod](https://www.gitpod.io/) is a tool integrating with Github that lets users develop in an online IDE whitin a docker-configured environment.
It is getting traction and offering more and more support of languages and tools.

Having such an environment configured for _exercism/java_ - and surely other languages - would help a lot any contibutors.
For newcomers, it saves the installation process - that may vary from installing java, gradle or only configlet. For seasoned contributors, it saves from maintaining the development environment into many personal machines.

Because exercism problems are often simple projects, there are no real need for a full-blown IDE, making Gitpod a good fit. I don't know if this could be applied to users, offering them a working environment for practice.

### PR DESCRIPTION
**Pull request by Kineolyan**
This PR offers an integration to Gitpod for all contributors.
Gitpod already comes with great support for java: sdkman + theia extensions already shipped. In addition to this, this PR adds:
 - `configlet` preinstalled
 - [the problem specification repository](https://github.com/exercism/problem-specifications) available in _/home/gitpod/specs_ and kept up-to-date
 - `configlet` tasks to implement a new exercise or update the readmes
 - standard java tasks: checkstyle, unit tests and journey execution

According to [Gitpod configuration](https://www.gitpod.io/docs/prebuilds/#when-a-prebuild-is-run), the workspace shoud be automatically build on merges to master.

Fixes #1835 

---

Reviewer Resources:

[Track Policies](https://github.com/exercism/java/blob/master/POLICIES.md#event-checklist)

### CODE REVIEW

**Comment by Kineolyan** [comment]
Note that I am assigned to #1815 . If needed, I am prepared to make some adjustments to the gitpod definition from my experience using it for this issue.

**Comment by jmrunkle** [comment]
Can you trim out the "Update master from upstream." commit? It is cluttering your file modifications making it hard to see what you actually changed. It might require you to rebase...

File: .gitpod.Dockerfile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+FROM gitpod/workspace-full
+
+USER gitpod
+
+RUN bash -c ". /home/gitpod/.sdkman/bin/sdkman-init.sh \
+    && sdk install java 11.0.7.fx-zulu \
+    && mkdir -p /home/gitpod/bin \
+    && wget -q -O - 'https://github.com/exercism/configlet/releases/download/v3.9.2/configlet-linux-64bit.tgz' | tar -xzf - -C /home/gitpod/bin \
```
Comment (jmrunkle) [inline .gitpod.Dockerfile:8]: This basically duplicates the bin/fetch_configlet.sh functionality, but picks a specific version. Just use fetch_configlet.

File: .gitpod.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    && git clone https://github.com/exercism/problem-specifications /home/gitpod/specs"
```
Comment (jmrunkle) [inline .gitpod.Dockerfile:9]: If you are going to do this at all, call the directory problem-specifications to match the style.

**Comment by jmrunkle** [comment]
If this PR is not updated or commented in in the next week, I plan to close it out. It looks like minimally it requires considerable merging at this point. If you add a comment saying your intentions for this PR I can leave it open.

**Comment by Kineolyan** [comment]
> It looks like minimally it requires considerable merging at this point.
Ok, for that amount of work, it may not be worth. Moreover, I know that Gitpod recently switch from Theia to VS Code, so it involves a lot of changes on that part as well.
Would it have been a feature upvoted many times, I would consider completing it. Since other contributors are fine with the current status, I am also in favor of abandoning this PR. Hence, I am closing this.

Cheers

**Pull request closed without merging**